test_main.py

```diff
@@ -47,15 +51,16 @@ def check_url_in_cache(url):
     """Check whether the URL is sourced from storage or not."""
     url_encoded = quote(url)
     cache_redirector_url = os.environ["KCIDB_CACHE_REDIRECTOR_URL"]
+    print(f"{cache_redirector_url}?{url_encoded}")
     response = requests.get(
         f"{cache_redirector_url}?{url_encoded}",
         timeout=10,   # Time in secs
-        allow_redirects=True
+        allow_redirects=False
     )
-    redirect_url = response.url
-    if response.status_code == 200:
+    if response.status_code == 302:
         # Check if the redirect URL matches the blob storage URL pattern
-        if redirect_url.startswith('https://storage.googleapis.com/'):
+        if response.headers.get("Location", ""). \
+            startswith('https://storage.googleapis.com/'):
```

Comment:
flake8 will probably complain about this indent, once you enable it.

Reply:
Could you please move the change to this function to its own commit, as we're essentially fixing it, and that's worth doing independently of the test change.

Reply:
okay